guppylang/checker/cfg_checker.py

```diff
@@ -191,6 +190,13 @@ class TypeHint(Note):
         span_label: ClassVar[str] = "This is of type `{ty}`"
         ty: Type
 
+    @dataclass(frozen=True)
+    class GlobalHint(Note):
+        message: ClassVar[str] = (
+            "{ident} may be shadowing a global name with type `{ty}` on some branches"
```

Comment:
If you also strore the definition in the note, you can use `defn.describe` instead of "name". Then it will explicitly say "function", "const", etc.

Reply:
As far as I can see only places have a describe method?

Reply:
Yes it should be `Definition.description`, sorry